### PR TITLE
Chore: Add value placeholders & preview for badge generator

### PR DESCRIPTION
**Pull request by chakflying**
⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md#can-i-create-a-pull-request-for-uptime-kuma

Tick the checkbox if you understand [x]: 
- [x] I have read and understand the pull request rules.

# Description

Fixes comment in #2915

New package "vite-plugin-commonjs" is added to allow for import of the server `config.js` file.

Also fixes bugs:

- cert-exp badge does not work because the color is set incorrectly.
- the "Badge Prefix/Suffix" does not actually describe what it does.

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- User interface (UI)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas
  (including JSDoc for methods)
- [x] My changes generate no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

## Screenshots (if any)

image

### CODE REVIEW

File: src/lang/en.json [hunk 1]
```diff
@@ -736,8 +736,8 @@
     "Badge Type": "Badge Type",
     "Badge Duration": "Badge Duration",
     "Badge Label": "Badge Label",
-    "Badge Prefix": "Badge Prefix",
-    "Badge Suffix": "Badge Suffix",
+    "Badge Prefix": "Badge Value Prefix",
+    "Badge Suffix": "Badge Value Suffix",
```
Comment (CommanderStorm) [inline src/lang/en.json:740]: For me, this is not clearer with the added value.
Imo, the current translation is good.

```suggestion
    "Badge Prefix": "Badge Prefix",
    "Badge Suffix": "Badge Suffix",
```

File: src/components/BadgeGeneratorDialog.vue [hunk 2]
```diff
@@ -109,7 +109,11 @@
 
                     <div class="mb-3">
                         <label for="value" class="form-label">{{ $t("Badge value (For Testing only.)") }}</label>
-                        <input id="value" v-model="badge.value" type="text" class="form-control" required>
+                        <input id="value" v-model="badge.value" type="text" class="form-control">
+                    </div>
+
+                    <div class="mb-3 pt-3 d-flex justify-content-center">
+                        <img :src="badgeURL" alt="Badge Preview">
```
Comment (CommanderStorm) [inline src/components/BadgeGeneratorDialog.vue:116]: In general, I think the preview seems a bit out of place (not a designer, feel free to overrule)

I think the following placements could be better:
- either next to the Badge-URL
- below this with a label `Preview`

image

File: src/components/BadgeGeneratorDialog.vue [hunk 3]
```diff
@@ -22,78 +22,78 @@
...
                     <div v-if=" (parameters[badge.type || 'null'] || [] ).includes('prefix') " class="mb-3">
                         <label for="prefix" class="form-label">{{ $t("Badge Prefix") }}</label>
-                        <input id="prefix" v-model="badge.prefix" type="text" class="form-control" required>
+                        <input id="prefix" v-model="badge.prefix" type="text" class="form-control">
                     </div>
 
                     <div v-if=" (parameters[badge.type || 'null'] || [] ).includes('suffix') " class="mb-3">
                         <label for="suffix" class="form-label">{{ $t("Badge Suffix") }}</label>
-                        <input id="suffix" v-model="badge.suffix" type="text" class="form-control" required>
+                        <input id="suffix" v-model="badge.suffix" type="text" placeholder="%" class="form-control">
```
[the remark this answers is not in the document]
Comment (chakflying) [reply src/components/BadgeGeneratorDialog.vue:41]: This is the "%" in the "100%" uptime.

File: src/lang/en.json (hunk 1 above)
Comment (chakflying) [reply src/lang/en.json:740]: "Badge Prefix" literally something in front of the badge. But that's not what it does.

image

image

"Badge Value Prefix" describes that the prefix is for the value of the badge.

File: src/lang/en.json [hunk 4]
```diff
@@ -736,8 +736,8 @@
...
     "Badge Label Color": "Badge Label Color",
     "Badge Color": "Badge Color",
     "Badge Label Prefix": "Badge Label Prefix",
```
Comment (kiznick) [inline src/lang/en.json:743]:
```suggestion
    "Badge Label Prefix": "Badge Label Prefix",
    "Badge Preview": "Badge Preview",
```

Missing i18n key.

File: src/lang/en.json [hunk 5]
```diff
@@ -736,8 +736,8 @@
     "Badge Type": "Badge Type",
     "Badge Duration": "Badge Duration",
```
Comment (kiznick) [inline src/lang/en.json:737]:
```suggestion
    "Badge Duration (in hours)": "Badge Duration (in hours)",
```

Change i18n key.

**Review by kiznick — CHANGES_REQUESTED** [review]
in en.json have some i18n missing keys and i18n keys that have changed.

File: src/components/BadgeGeneratorDialog.vue
Comment (kiznick) [inline src/components/BadgeGeneratorDialog.vue:1]: In line `121`, can we remove `id` from 
`<CopyableInput id="push-url" v-model="badgeURL" type="url" disabled="disabled" />`
or change it to `badge-url`?

**Comment by CommanderStorm** [comment]
@louislam
I think this could be a valuable addition to the [1.22.1 milestone](https://github.com/louislam/uptime-kuma/milestone/35).
What do you think?

**Comment by louislam** [comment]
> @louislam I think this could be a valuable addition to the [1.22.1 milestone](https://github.com/louislam/uptime-kuma/milestone/35). What do you think?

No problem, it is indeed hard to use without knowing the default value.

**Review by kiznick — APPROVED** [review]

**Comment by louislam** [comment]
> @louislam I think this could be a valuable addition to the [1.22.1 milestone](https://github.com/louislam/uptime-kuma/milestone/35). What do you think?

This branch is not based on the 1.22.X branch, so it has to be in 1.23.0.

**Pull request merged**